exist-core/src/test/java/org/exist/http/RESTServiceTest.java

```diff
@@ -692,6 +705,308 @@ public void storedQuery() throws IOException {
+        try {
+            connect.setRequestMethod("GET");
+            connect.connect();
+
+            final int r = connect.getResponseCode();
+            assertEquals("Server returned response code " + r, HttpStatus.OK_200, r);
+
+            final String response = readResponse(connect.getInputStream());
+
+            final Source expectedSource = Input.from(
+                    "<authorization>\n" +
+                            "    <sm:id xmlns:sm=\"http://exist-db.org/xquery/securitymanager\">\n" +
+                            "        <sm:real>\n" +
+                            "            <sm:username>guest</sm:username>\n" +
+                            "            <sm:groups>\n" +
+                            "                <sm:group>guest</sm:group>\n" +
+                            "            </sm:groups>\n" +
+                            "        </sm:real>\n" +
+                            "    </sm:id>\n" +
+                            "    <header/>\n" +
+                            "</authorization>").build();
+            final Source actualSource = Input.from(response).build();
+
+            final Diff diff = DiffBuilder.compare(expectedSource)
+                    .withTest(actualSource)
+                    .checkForSimilar()
+                    .build();
+
+            assertFalse(diff.toString(), diff.hasDifferences());
+
+        } finally {
+            connect.disconnect();
+        }
+    }
+
+    @Test
+    public void execQueryWithBasicAuth() throws IOException {
+        doPut(AUTH_QUERY, "auth.xq", HttpStatus.CREATED_201);
+
+        // allow query to be executed only by owner
+        chmod(XmldbURI.ROOT_COLLECTION + "/test/auth.xq", "rwxrw-r--");
+
+        // call the auth.xq
+        final String uri = getCollectionUri() + "/auth.xq";
+        final HttpURLConnection connect = getConnection(uri);
+        try {
+            connect.setRequestProperty("Authorization", "Basic " + credentials);
+            connect.setRequestMethod("GET");
+            connect.connect();
+
+            final int r = connect.getResponseCode();
+            assertEquals("Server returned response code " + r, HttpStatus.OK_200, r);
+
+            final String response = readResponse(connect.getInputStream());
+
+            final Source expectedSource = Input.from(
+                    "<authorization>\n" +
+                            "    <sm:id xmlns:sm=\"http://exist-db.org/xquery/securitymanager\">\n" +
+                            "        <sm:real>\n" +
+                            "            <sm:username>admin</sm:username>\n" +
+                            "            <sm:groups>\n" +
+                            "                <sm:group>dba</sm:group>\n" +
+                            "            </sm:groups>\n" +
+                            "        </sm:real>\n" +
+                            "    </sm:id>\n" +
+                            "    <header>Basic YWRtaW46</header>\n" +
```

Comment:
Here it would make sense to change the case of the header to something like `BaSiC` to make sure that the comparison is done specification compliant.

Reply:
@reinhapa I have now added a test specifically for that: `RESTServiceTest#execQueryWithBasicAuthCaseInsensitive()`